src/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,494 @@
+import org.zaproxy.zap.model.TechSet;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+/**
+ * The MongoInjection plugin identifies MongoDB injection vulnerabilities
+ *
+ * @author l.casciaro
+ */
+public class MongoDbInjection extends AbstractAppParamPlugin {
+
+	// Prefix for internationalised messages used by this rule
+	private static final String MESSAGE_PREFIX = "ascanalpha.mongodb.";
+	// Constants	
+	private static final String ALL_DATA_ATTACK = "alldata";
+	private static final String CRASH_ATTACK = "crash";
+	private static final String SLEEP_ATTACK ="sleep";
+	private static final String JSON_ATTACK = "json";
+	private static final String AUTH_BYPASS_ATTACK = "authbypass";
+	private static final String URL_TOKEN ="[$ZAP]";
+	private static final String JSON_TOKEN = "$ZAP";
+	// Variables
+	private boolean isJsonPayload;
+	private boolean doAllDataScan, doCrashScan, doTimedScan, doJsonScan, getMoreConfidence, doAuthBypass;
+	// mongo sleep argument: (1000 = 3 sec)
+	private static int SLEEP_TIME_SHORT;
+	private static int SLEEP_TIME_LONG;
+	private static final int DELTA_TIME=1500;
+	private static final int RTT_MAX = (int) TimeUnit.SECONDS.toMillis(( Model.getSingleton().getOptionsParam().
+											 getConnectionParam().getTimeoutInSecs()-5));
+	// Packages of attack rules
+	private static final String[] ALL_DATA_PARAM_INJECTION = new String[] {"[$ne]", "[$regex]", "[$gt]"};
+	private static final String[] ALL_DATA_VALUE_INJECTION = new String[]  {"", ".*", "0"};
+	private static final String[] CRASH_INJECTION = new String[] {"\"", "'", "//", "});",");"};
+	private static final String INSERT_TIME_SHORT = "INSERT_TIME_SHORT";
+	private static final String INSERT_TIME_LONG = "INSERT_TIME_LONG";
+	private static final String[][] SLEEP_INJECTION = {
+			// Attacking where and group clauses
+			{"'); sleep("+INSERT_TIME_SHORT+"); print('","'); sleep("+INSERT_TIME_LONG+"); print('"}, 
+			{"'; sleep("+INSERT_TIME_SHORT+"); var x='", "'; sleep("+INSERT_TIME_LONG+"); var x='"},
+			{"; sleep("+INSERT_TIME_SHORT+")", "; sleep("+INSERT_TIME_LONG+")"},
+			{"zap); sleep("+INSERT_TIME_SHORT,"zap); sleep("+INSERT_TIME_LONG },
+			// Attacking mapReduce clause
+			{"_id);}, function(inj) { sleep("+INSERT_TIME_SHORT+");return 1;}, { out: 'x'}); "+
+			"db.injection.mapReduce(function() { emit(1,1",	
+			"_id);}, function(inj) { sleep("+INSERT_TIME_LONG+"); return 1;}, { out: 'x'}); "+
+			"db.injection.mapReduce(function() { emit(1,1"}};
+	private static final String[][] JSON_INJECTION = {{"$ne", "0"}, {"$gt", ""}, {"$regex", ".*"}};
+	// Error messages that addressing to a well-known vulnerability
+	private  final Pattern[] errorPatterns = {
+			Pattern.compile("RuntimeException: SyntaxError: unterminated string literal", Pattern.CASE_INSENSITIVE),
+			Pattern.compile("MongoResultException", Pattern.CASE_INSENSITIVE)};
+	// Log prints
+	private static final String JSON_EX_LOG = "try to convert the payload in json format";

```

Comment:
I think these should be "trying" so that the full message ends up being: `Caught _______ when trying to convert the payload` etc.